Tasks/Common/packaging-common/cache/feedUtilities.ts

```diff
@@ -0,0 +1,58 @@
+import * as pkgLocationUtils from "../locationUtilities";
+import * as tl from "azure-pipelines-task-lib";
+import Axios, { AxiosRequestConfig } from "axios";
+
+interface IPackage {
+  id: string;
+  name: string;
+  version: string;
+}
+
+export async function doesPackageExist(hash: string): Promise<boolean> {
+  const feedId = tl.getInput("feedList");
+  // Getting package name from hash
+  const packageId = tl
+    .getVariable("Build.DefinitionName")
+    .replace(/\s/g, "")
+    .substring(0, 255)
+    .toLowerCase();
+
+  const version = `1.0.0-${hash}`;
+  console.log(version);
+
+  const accessToken = pkgLocationUtils.getSystemAccessToken();
+  const collectionUri = process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI;
+
+  let instance: string = "";
+  const legacyRegex = /https:\/\/(\S+).visualstudio.com\S+/g;
+  const newRegex = /https:\/\/dev.azure.com\/(\S+)\//g;
+
+  const legacyUrl = legacyRegex.exec(collectionUri);
+  const newUrl = newRegex.exec(collectionUri);
+
+  if (legacyUrl) {
+    instance = legacyUrl[1];
+  } else if (newUrl) {
+    instance = newUrl[1];
+  }
```

Comment:
Else throw?

Reply:
Good idea. The accompanying change to this is to wrap the calling of `doesPackageExist` in a try/catch, so that the output variables of the task don't allow for subsequent caching tasks to interpret a cache hit/miss.